rrtmgp/kernels-openacc/mo_gas_optics_kernels.F90

```diff
@@ -214,27 +222,41 @@ subroutine compute_tau_absorption(                &
     ! ----------------------------------------------------------------
 
     !$acc enter data create(itropo_lower, itropo_upper)
+    !$omp target enter data map(alloc:itropo_lower, itropo_upper)
     !$acc enter data copyin(play, tlay, tropo, gpoint_flavor, jeta, jtemp, col_gas, fminor, tau)
+    !$omp target enter data map(to:play, tlay, tropo, gpoint_flavor, jeta, jtemp, col_gas, fminor, tau)
 
     ! ---------------------
     ! Layer limits of upper, lower atmospheres
     ! ---------------------
     top_at_1 = play(1,1) < play(1, nlay)
     if(top_at_1) then
       !$acc parallel loop
+      !$omp target teams distribute parallel do simd
       do icol = 1,ncol
         itropo_lower(icol,2) = nlay
+#ifdef _CRAYFTN
```

Comment:
Not pretty but effective. Do you know if Cray plans to support the `minloc` and `maxloc` intrinsics on the device at some point?

Reply:
I made the request almost a year ago, but sadly it has not been implemented yet.

I have made this request to other vendors like Intel and they will support this early on.